src/java.base/share/classes/java/io/DataInputStream.java

```diff
@@ -54,6 +56,8 @@ public DataInputStream(InputStream in) {
         super(in);
     }
 
+    private final byte[] readBuffer = new byte[8];
```

Comment:
No objection to the readBuffer but it does make it wonder if we should re-visit bytearr and chararr as they are only need for reading modified UTF-8 strings and shouldn't need to be eagerly created.

Reply:
https://bugs.openjdk.org/browse/JDK-8300979